experiments/src/browsergym/experiments/loop.py

```diff
@@ -180,17 +193,37 @@ def run(self):
             self.err_msg = err_msg
             self.stack_trace = stack_trace
 
-            logging.warning(err_msg + "\n" + stack_trace)
+            logger.warning(err_msg + "\n" + stack_trace)
             if _is_debugging() and self.enable_debug:
                 raise
 
         finally:
             try:
                 step_info.save_step_info(self.exp_dir)
+            except Exception as e:
+                logger.error(f"Error while saving step info in the finally block: {e}")
+            try:
                 _save_summary_info(episode_info, self.exp_dir, err_msg, stack_trace)
+            except Exception as e:
+                logger.error(f"Error while saving summary info in the finally block: {e}")
+            try:
                 env.close()
             except Exception as e:
-                logging.error(f"Error while finalizing the experiment loop: {e}")
+                logger.error(f"Error while closing the environment in the finally block: {e}")
+
+    def get_logger(self):
+        logger = logging.getLogger()
+        logger.setLevel(self.logging_level)
+        file_handler = logging.FileHandler(self.exp_dir / "experiment.log")
+        file_handler.setLevel(self.logging_level)
+        formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+        file_handler.setFormatter(formatter)
+        logger.addHandler(file_handler)
+
+        if self.logging_level < logging.INFO:
+            logging.getLogger("openai._base_client").setLevel(logging.INFO)
```

Comment:
This is openai specific, ideally shouldn't be here. Does this work if openai is not installed?

Reply:
I know, but I think it's important. otherwise, it puts in the log all prompt that we send.
It's not going to crash if it's not installed. And if we need to add other filters from other packages, we can add them here.

Reply:
Okey dokey! If it doesn't crash then all good